src/main/java/hudson/plugins/scm/koji/model/RPM.java

```diff
@@ -62,11 +63,54 @@ public String getArch() {
 
     @Override
     public String toString() {
-        return nvr + '.' + arch + ".suffix";
+        if (Suffix.INSTANCE.endsWithSuffix(nvr)) {
+            return nvr;
+        }
+        Suffix suffix = Suffix.INSTANCE;
+        return nvr + '.' + arch + '.' + suffix.getSuffix(url);
     }
 
     public String getFilename(String suffix) {
         return Optional.ofNullable(filename).orElse(nvr + '.' + arch + "." + suffix);
     }
 
+    public String getUrl() {
+        return url;
+    }
+
+    public void setUrl(String url) {
+        this.url = url;
+    }
+
+    public boolean hasUrl() {
+        return url != null;
+    }
+
+    public static enum Suffix {
```

Comment:
`static` is useless with enum

Reply:
judovana wanted me to leave it that way.